yarn-project/key-store/src/test_key_store.ts

```diff
@@ -58,19 +58,26 @@ export class TestKeyStore implements KeyStore {
     const accountAddressFr = poseidon2Hash([publicKeysHash, partialAddress, GeneratorIndex.CONTRACT_ADDRESS_V1]);
     const accountAddress = AztecAddress.fromField(accountAddressFr);
 
-    // We save the keys to db
+    // We save the keys to db associated with the account address
     await this.#keys.set(`${accountAddress.toString()}-public_keys_hash`, publicKeysHash.toBuffer());
 
-    await this.#keys.set(`${accountAddress.toString()}-nsk_m`, masterNullifierSecretKey.toBuffer());
+    await this.#keys.set(`${accountAddress.toString()}-nsk_m-account`, masterNullifierSecretKey.toBuffer());
     await this.#keys.set(`${accountAddress.toString()}-ivsk_m`, masterIncomingViewingSecretKey.toBuffer());
     await this.#keys.set(`${accountAddress.toString()}-ovsk_m`, masterOutgoingViewingSecretKey.toBuffer());
     await this.#keys.set(`${accountAddress.toString()}-tsk_m`, masterTaggingSecretKey.toBuffer());
 
-    await this.#keys.set(`${accountAddress.toString()}-npk_m`, masterNullifierPublicKey.toBuffer());
+    await this.#keys.set(`${accountAddress.toString()}-npk_m-account`, masterNullifierPublicKey.toBuffer());
     await this.#keys.set(`${accountAddress.toString()}-ivpk_m`, masterIncomingViewingPublicKey.toBuffer());
     await this.#keys.set(`${accountAddress.toString()}-ovpk_m`, masterOutgoingViewingPublicKey.toBuffer());
     await this.#keys.set(`${accountAddress.toString()}-tpk_m`, masterTaggingPublicKey.toBuffer());
 
+    const masterNullifierPublicKeyHash = poseidon2Hash(masterNullifierPublicKey.toFields());
+
+    // We save nullifier keys and account address to db under the master nullifier key hash
+    await this.#keys.set(`${masterNullifierPublicKeyHash.toString()}-nsk_m-mnpkh`, masterNullifierSecretKey.toBuffer());
```

Comment:
Not sure I like the `mnpkh` name here as everything else is using `_m` as the master it is a bit weird to read when it is now a prefix. Generally I don't like the lack of consistency of what the naming means here. I think if it is required with more information here, it should include a `from-to` type of declaration. 

For example:
```typescript
await this.#keys.set(`${accountAddress.toString()}-address-npk_m`, masterNullifierPublicKey.toBuffer());
await this.#keys.set(`${accountAddress.toString()}-address-ivpk_m`, masterIncomingViewingPublicKey.toBuffer());
await this.#keys.set(`${accountAddress.toString()}-address-ovpk_m`, masterOutgoingViewingPublicKey.toBuffer());
await this.#keys.set(`${accountAddress.toString()}-address-tpk_m`, masterTaggingPublicKey.toBuffer());

const masterNullifierPublicKeyHash = poseidon2Hash(masterNullifierPublicKey.toFields());
// We save nullifier keys and account address to db under the master nullifier key hash
await this.#keys.set(`${masterNullifierPublicKeyHash.toString()}-npkh_m-nsk_m`, masterNullifierSecretKey.toBuffer());
```